.gitlab-ci.yml

```diff
@@ -1,67 +1,39 @@
-#   For CUDA version Major.X (where X is the latest patch of the major version) distributed round robin
-#   for each compiler within the file.
-#   There are exceptions for intermediate minor versions of CUDA
-#   stageRun0 = all CUDA version Major.0
-#   stageRun1 = all CUDA where minor != 0 and not latest minor version
-# Compile Tests
-#   stageCompileN = round robin distributed within the file
+# generate
+#  creates the actual test jobs via python script
+# run-tests
+#  execute the generated test jobs
+#  the ordering is mainly random
+#  the python generator can schedule some jobs on specific positions
 
 stages:
-  - stageCompile0
-  - stageRun0
-  - stageCompile1
-  - stageRun1
+  - generator
+  - run-test-jobs
 
 
 variables:
-  ALPAKA_GITLAB_CI_CONTAINER_VERSION: "1.4"
-  ALPAKA_CI_OS_NAME: "Linux"
-  alpaka_ACC_CPU_B_SEQ_T_SEQ_ENABLE: "ON"
-  alpaka_ACC_CPU_B_SEQ_T_THREADS_ENABLE: "ON"
-  alpaka_ACC_CPU_B_SEQ_T_FIBERS_ENABLE: "OFF"
-  alpaka_ACC_CPU_B_TBB_T_SEQ_ENABLE: "OFF"
-  alpaka_ACC_CPU_B_OMP2_T_SEQ_ENABLE: "ON"
-  alpaka_ACC_CPU_B_SEQ_T_OMP2_ENABLE: "ON"
-  alpaka_ACC_ANY_BT_OMP5_ENABLE: "OFF"
-  alpaka_ACC_ANY_BT_OACC_ENABLE: "OFF"
-  alpaka_ACC_GPU_CUDA_ENABLE: "OFF"
-  alpaka_ACC_GPU_CUDA_ONLY_MODE: "OFF"
-  alpaka_ACC_GPU_HIP_ENABLE: "OFF"
-  alpaka_ACC_GPU_HIP_ONLY_MODE: "OFF"
-  # If ALPAKA_CI_ANALYSIS is OFF compile and execute runtime tests else compile only.
-  ALPAKA_CI_ANALYSIS: "OFF"
-  ALPAKA_CI_RUN_TESTS: "ON"
-  alpaka_CI: GITLAB
-  # needs to be enabled, that test on the GPU are executed
-  ALPAKA_FORCE_RUNTIME_TEST: "ON"
-  ALPAKA_CI_SANITIZERS: ""
-  ALPAKA_CI_INSTALL_CUDA: "OFF"
-  ALPAKA_CI_INSTALL_HIP: "OFF"
-  ALPAKA_CI_CMAKE_DIR: "$HOME/cmake"
-  BOOST_ROOT: "$HOME/boost"
-  ALPAKA_CI_BOOST_LIB_DIR: "$HOME/boost_libs"
-  ALPAKA_CI_CUDA_DIR: "$HOME/cuda"
-  ALPAKA_CI_HIP_ROOT_DIR: "$HOME/hip"
+  # container version of the generated jobs
+  # should be merged with ALPAKA_GITLAB_CI_CONTAINER_VERSION
+  # see: script/job_generator/generate_job_yaml.py
+  ALPAKA_GITLAB_CI_GENERATOR_CONTAINER_VERSION: "3.0"
 
-include:
-  - local: '/script/gitlabci/job_base.yml'
-  - local: '/script/gitlabci/job_cuda9.2.yml'
-  - local: '/script/gitlabci/job_cuda10.0.yml'
-  - local: '/script/gitlabci/job_cuda10.1.yml'
-  - local: '/script/gitlabci/job_cuda10.2.yml'
-  - local: '/script/gitlabci/job_cuda11.0.yml'
-  - local: '/script/gitlabci/job_cuda11.1.yml'
-  - local: '/script/gitlabci/job_cuda11.2.yml'
-  - local: '/script/gitlabci/job_cuda11.3.yml'
-  - local: '/script/gitlabci/job_cuda11.4.yml'
-  - local: '/script/gitlabci/job_cuda11.5.yml'
-  - local: '/script/gitlabci/job_cuda11.6.yml'
-  - local: '/script/gitlabci/job_hip4.2.yml'
-  - local: '/script/gitlabci/job_hip4.3.yml'
-  - local: '/script/gitlabci/job_hip4.5.yml'
-  - local: '/script/gitlabci/job_hip5.0.yml'
-  - local: '/script/gitlabci/job_hip5.1.yml'
-  - local: '/script/gitlabci/job_hip5.2.yml'
+generate:
+  stage: generator
+  image: alpine:latest
```

Comment:
AFAIK most of the CI runs on Ubuntu. Why do we need alpine for this job? Could this run on Ubuntu as well?

Reply:
Because the alpine image is much smaller, than the ubuntu image. The job only generates a yaml file for the following up jobs. And the only difference in the job script compare to use an ubuntu image is, that we use `apk` instead `apt`.

Reply:
ok then!

Reply:
Is a fixed alpine version maybe better to avoid that if the image is switching to a new python version that our script stops working?

Reply:
Better I fixed the Python version itself.